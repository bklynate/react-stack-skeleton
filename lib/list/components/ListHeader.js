"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = ListHeader;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The footer for the todo list.
 */
function ListHeader(_ref) {
  var newItemValue = _ref.newItemValue,
      onChangeNewItem = _ref.onChangeNewItem,
      otherProps = _objectWithoutProperties(_ref, ["newItemValue", "onChangeNewItem"]);

  return _react2.default.createElement(
    "header",
    _extends({ className: "header" }, otherProps),
    _react2.default.createElement(
      "h1",
      null,
      "todos"
    ),
    _react2.default.createElement("input", {
      className: "new-todo",
      placeholder: "What needs to be done?",
      value: newItemValue,
      onChange: onChangeNewItem,
      autoFocus: true
    })
  );
}