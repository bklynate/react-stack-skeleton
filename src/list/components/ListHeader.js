/* @flow */
import React from 'react';

type Props = {
  newItemValue: string,
  onChangeNewItem: (e: SyntheticEvent) => void,
};

/**
 * The footer for the todo list.
 */
export default function ListHeader({ newItemValue, onChangeNewItem, ...otherProps }: *): React.Element<*> {
  return (
   <header className="header" {...otherProps}>
     <h1>todos</h1>
     <input
       className="new-todo"
       placeholder="What needs to be done?"
       value={newItemValue}
       onChange={onChangeNewItem}
       autoFocus
     />
   </header>
  );
}
