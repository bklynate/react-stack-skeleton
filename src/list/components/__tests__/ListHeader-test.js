import React from 'react';
import renderer from 'react-test-renderer';

import ListHeader from '../ListHeader';

test('it displays the given value for the new item.', () => {
  const component = renderer.create(
    <ListHeader newItemValue="Test Value" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it calls the onChangeNewItem function if the item is changed.', () => {
  const onChangeFn = jest.fn();
  const component = renderer.create(
    <ListHeader onChangeNewItem={onChangeFn} />
  );
  const tree = component.toJSON();
  const newItemInput = tree.children[1];

  newItemInput.props.onChange();
  expect(onChangeFn).toHaveBeenCalledTimes(1);
});
