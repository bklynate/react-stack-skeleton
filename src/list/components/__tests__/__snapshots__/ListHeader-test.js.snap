exports[`test it displays the given value for the new item. 1`] = `
<header
  className="header">
  <h1>
    todos
  </h1>
  <input
    autoFocus={true}
    className="new-todo"
    onChange={undefined}
    placeholder="What needs to be done?"
    value="Test Value" />
</header>
`;
